components/director/pkg/jsonschema/validator.go

```diff
@@ -45,30 +47,75 @@ func NewValidatorFromRawSchema(jsonSchema interface{}) (*validator, error) {
 	}, nil
 }
 
-func (v *validator) ValidateString(json string) (bool, error) {
+func (v *validator) ValidateString(json string) (ValidationResult, error) {
 	if v.schema == nil {
-		return true, nil
+		return ValidationResult{
+			Valid: true,
+			Error: nil,
+		}, nil
 	}
 
 	jsonLoader := gojsonschema.NewStringLoader(json)
 	result, err := v.schema.Validate(jsonLoader)
 	if err != nil {
-		return false, err
+		return ValidationResult{
+			Valid: false,
+			Error: nil,
+		}, err
 	}
 
-	return result.Valid(), nil
+	var validationError error
+	if len(result.Errors()) > 0 {
```

Comment:
to discuss: using mutlierror, for example `https://github.com/uber-go/multierr/blob/master/error.go`